### DISCUSSION TITLE
Creating custom GPTs in open webui

### DISCUSSION THREAD

**Discussion by gqoew** [open-webui/open-webui#3646] [General]
GPTs are custom versions of ChatGPT that users can tailor for specific tasks or topics by combining instructions, knowledge, and capabilities.

Do you have any plans to add this GPT building capability in open webui?

**Comment by gqoew** [comment]
I managed to create a "GPT" by creating a new "model" under the "workspace" tab.

**Reply by lukestanley** [reply]
But can you provide it with an JSON OpenAPI / Swagger spec for it to do function calling on, in the same way?
@gqoew

**Reply by erickpeniche** [reply]
> But can you provide it with an JSON OpenAPI / Swagger spec for it to do function calling on, in the same way? @gqoew

@lukestanley did you ever find out if this is possible? Looking to implement the same in a custom model

**Reply by lukestanley** [reply]
I didn't yet but I'd search for OpenAPI in the code or ask GitHub Copilot to.

**Reply by lukestanley** [reply]
I just asked it and it looks like it couldn't find OpenAPI mentioned, but it does allow arbitrary Python code execution, so we shouldn't be too hard to do all kinds of fun things.

**Comment by doubianimehdi** [comment]
Hi , I know it's closed but i'm looking to have an interface that allows to create custom gpts as easily as chatgpt and/or perplexity's spaces ... i find the current possibilities too complex for average users ... is there a plan to make it as lean and easy ? thank you !
